pkg/products/threescale/reconciler.go

```diff
@@ -390,30 +390,46 @@ func (r *Reconciler) Reconcile(ctx context.Context, installation *integreatlyv1a
 		return phase, err
 	}
 
-	ingressRouterService, err := customDomain.GetIngressRouterService(ctx, serverClient)
-	if err != nil || len(ingressRouterService.Status.LoadBalancer.Ingress) == 0 {
-		errorMessage := "failed to retrieve ingress router service"
-		r.log.Error(errorMessage, err)
-		phase = integreatlyv1alpha1.PhaseFailed
-		events.HandleError(r.recorder, installation, phase, errorMessage, err)
-		return phase, fmt.Errorf("%s: %v", errorMessage, err)
+	threescaleMasterRoute, err := r.getThreescaleRoute(ctx, serverClient, "system-master", nil)
+	if err != nil || threescaleMasterRoute == nil {
+		return integreatlyv1alpha1.PhaseFailed, err
 	}
 
-	ips, err := customDomain.GetIngressRouterIPs(ingressRouterService.Status.LoadBalancer.Ingress)
-	if err != nil {
-		errorMessage := "failed to retrieve ingress router ips"
-		r.log.Error(errorMessage, err)
-		phase = integreatlyv1alpha1.PhaseFailed
-		events.HandleError(r.recorder, installation, phase, errorMessage, err)
-		return phase, fmt.Errorf("%s: %v", errorMessage, err)
+	listOfIngresServicesNames := []string{"router-default"}
+	if customDomainActive {
+		// get ingres custom domain service (LoadBalancer, router) name from 3scale route
+		if len(threescaleMasterRoute.Status.Ingress) > 1 {
```

Comment:
We can just take the first item in the array. If there are multiple Ingress that just means the route is on multiple. The first in the list should always work

Reply:
the first item - is default . It's always "router-default". Here we are checking if we have more then one router (len > 1), and in this case we are taking the second one, that is CustomDomain

Reply:
The first in the list - is always working

Reply:
Logic was moved to ping3scalePortals